server/src/com/thoughtworks/go/config/GoConfigDao.java

```diff
@@ -179,7 +179,7 @@ public String md5OfConfigFile() {
     }
 
     public void updatePipeline(PipelineConfig pipelineConfig, LocalizedOperationResult result, Username currentUser, PipelineConfigService.SaveCommand saveCommand) {
-        synchronized (writeLock) {
+        synchronized (goConfigWriteLock.mutex()) {
             if (saveCommand.hasWritePermissions()) {
```

Comment:
This internally calls GoConfigService#canEditPipeline which [updates the result](https://github.com/gocd/gocd/blob/4e9c713b7fb49acea4775279312bf36c4bb6fc70/server/src/com/thoughtworks/go/server/service/GoConfigService.java#L156) before returning a true/false value. Since the same method is also use for config save from UI, I did not change the behaviour.